spec/requests/sessions_spec.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+describe 'Sessions', type: :request do
+  let(:recaptcha_id) { 'recaptcha_form' }
+
+  describe '#new' do
+    after { cookies[:login_recaptcha] = nil }
+
+    context 'when recaptcha is not configured' do
+      before do
+        allow(RecaptchaService).to receive(:recaptcha_configured?).and_return(false)
+      end
+
+      it 'does not display recaptcha' do
+        cookies[:login_recaptcha] = true
+        get new_user_session_path
+
+        expect(response.body).to_not include recaptcha_id
+      end
+    end 
+
+    context 'when recaptcha is configured' do
+      before do
+        allow(RecaptchaService).to receive(:recaptcha_configured?).and_return(true)
+      end
+
+      context 'and user has recaptcha cookie' do
+        it 'displays recaptcha' do
+          cookies[:login_recaptcha] = true
+          get new_user_session_path
+
+          expect(response.body).to include recaptcha_id
+        end
+      end
+
+      context 'and user does not have recaptcha cookie' do
+        context 'and recaptcha is specified in query string' do
+          before do
+            get new_user_session_path, params: { recaptcha: true }
+          end
+
+          it 'displays recaptcha' do
+            expect(response.body).to include recaptcha_id
+          end
+
+          it 'sets recaptcha cookie' do
+            expect(cookies[:login_recaptcha]).to eq 'true'
```

Comment:
Why is `'true'` a string instead of a boolean?

Reply:
I think it's because cookies are always stored as strings, so even though the cookie is being stored with a boolean (i.e. `cookies.permanent[:login_recaptcha] = true`), it's being converted to a string. Maybe I should use "1" rather than "true"?

Reply:
This is fine to me then. It got the point across. I just wanted to make sure 👍